ee/tools/puppet/fleetdm/lib/puppet/util/fleet_client.rb

```diff
@@ -0,0 +1,99 @@
+require 'net/http'
+require 'uri'
+require 'json'
+require 'puppet'
+require 'hiera_puppet'
+
+module Puppet::Util
+  # FleetClient provides an interface for making HTTP requests to a Fleet server.
+  class FleetClient
+    def initialize(host, token)
+      @host = host
+      @token = token
+    end
+
+    # Pre-assigns a profile to a host. Note that the profile assignment is not
+    # effective until the sibling `match_profiles` method is called.
+    #
+    # @param uuid [String] The host uuid.
+    # @param profile_xml [String] Raw XML with the configuration profile.
+    # @param group [String] Used to construct a team name.
+    # @return [Hash] The response status code, headers, and body.
+    def preassign_profile(uuid, profile_xml, group)
+      post(
+        '/api/latest/fleet/mdm/apple/profiles/preassign',
+        {
+          'external_node_identifier' => Puppet[:node_name_value],
```

Comment:
Just to make sure, `:node_name_value` is unique per `host_uuid`, right? A 1:1 mapping of external node identifier to host uuid?